addons/l10n_de/models/datev.py

```diff
@@ -16,24 +16,22 @@ def _get_product_accounts(self):
         result = super(ProductTemplate, self)._get_product_accounts()
         company = self.env.company
         if company.account_fiscal_country_id.code == "DE":
-            if not self.property_account_income_id:
-                taxes = self.taxes_id.filtered_domain(self.env['account.tax']._check_company_domain(company))
-                if not result['income'] or (result['income'].tax_ids and taxes and taxes[0] not in result['income'].tax_ids):
```

Comment:
we seem to have removed this `or` condition 
`... or not result['income'].tax_ids & taxes[0]`